src/pager.js

```diff
@@ -1,122 +1,75 @@
+  @bindable({defaultBindingMode: bindingMode.twoWay}) page
+
+  @bindable data    //set the data if you want to alter the
+  @bindable range
+  @bindable limit
+  @bindable pages
+
+  constructor() {
+    this.page = this.page || 1;
   }
 
-  reloadCount() {
-    if (this.resource) {
-      return this._calculatePages();
-    }
+  dataChanged() {
+    this.page = 1;
+    this.pages = this.data.length;
+  }
 
-    this._calculateRange();
+  pagesChanged() {
+    this.page = 1;
   }
 
-  pageChanged(newValue, oldValue) {
-    if (newValue !== oldValue) {
-      this.goToPage(newValue);
-    }
+  next() {
+    return this.change(this.page + 1);
   }
 
-  criteriaChanged(newValue, oldValue) {
-    if (this.resource && (newValue !== oldValue)) {
-      this._calculatePages();
-    }
+  prev() {
+    return this.change(this.page - 1);
   }
 
-  nextPage() {
-    if (this.page < this.pages) {
-      this.page++;
+  change(page) {
+    if (this.page < 0) {
+      this.page = 0;
+      return this.page;
     }
-  }
 
-  prevPage() {
-    if (this.page > 1 && this.page <= this.pages) {
-      this.page--;
+    if (this.page > this.pages) {
+      this.page = this.pages;
+      return this.page;
     }
-  }
 
-  lastPage() {
-    this.page = this.pages;
-  }
+    this.page = page;
 
-  firstPage() {
-    this.page = 1;
+    return this.page;
   }
 
-  goToPage(page) {
-    if (page < 0 || page > this.pages) {
-      return;
+  @computedFrom('page', 'pages', 'range')
+  get numbers() {
+    let numbers = [];
+
+    for (let index = 0; index < this.range * 2 + 1; index++) {
+      numbers[index] = number.call(this, index);
     }
 
-    this._calculateRange();
-  }
+    return numbers;
 
-  /**
-   * caclulate the amount of pages to show
-   */
-  _calculateRange() {
-    let rangeStart = Math.max(this.page - this.pagerange, 1);
-    let rangeEnd   = Math.min(this.page + this.pagerange, this.pages);
-    let navs       = [];
-    let i;
-
-    if (this.page <= this.pagerange) {
-      rangeEnd = Math.min(this.pagerange * 2 + 1, this.pages);
-    }
+    function number(index) {
+      if (this.page + this.range > this.pages) {
+        return this.pages - (this.range * 2) + index;
+      }
 
-    if (this.page > (this.pages - this.pagerange)) {
-      if (this.pages <= (this.pagerange * 2)) {
-        rangeStart = 1;
-      } else {
-        rangeStart = Math.max(this.pages - this.pagerange * 2, this.pagerange);
+      if (this.page - this.range < 1) {
+        return index + 1;
       }
-    }
 
-    for (i = rangeStart; i < rangeEnd + 1; i++) {
-      navs.push({
-        text   : i.toString(),
-        current: i === this.page,
-        load   : page => {
-          this.page = parseInt(page);
-        }
-      });
+      return this.page - this.range + index;
     }
-
-    this.navs = navs;
   }
 
-  /**
-   * fetch data from DB or given array to calculate the amount of pages
-   */
-  _calculatePages() {
-    if (Array.isArray(this.resource)) {
```

Comment:
What if the `resource` is an array and not a repository? Now a repository is always needed.